File: jeanstationfrontend/src/app/models/Order.ts
import OrderProduct from "./OrderProduct";
import User from "../models/User"
import CartProduct from "./CartProduct";

export default class Order{
    id: number;
    orderId: string;
    userId: string;
    products: Array<CartProduct>;
    orderDateTime: Date;
    addressId: string;
    totalPrice: number;
    discountPrice: number;
    couponCode: string;
    orderStatus: string;
    user: User;

    constructor(){
        this.id = 0;
        this.orderId = "";
        this.userId = "";
        this.products = [];
        this.orderDateTime = new Date();
        this.addressId = "";
        this.totalPrice = 0;
        this.discountPrice = 0;
        this.couponCode = "";
        this.orderStatus = "";
        this.user = new User();
    }

}

File: NotificationsAPI/Models/Order.cs
using System;
using System.Collections.Generic;

namespace NotificationsAPI.Models
{
    public class Order
    {
        public int Id { get; set; }
        public string OrderId { get; set; }
        public string UserId { get; set; }
        public User User { get; set; }
        public List<OrderProduct> Products { get; set; }
        public DateTime OrderDatetime { get; set; }
        public string AddressId { get; set; }
        public float TotalPrice { get; set; }
        public float DiscountPrice { get; set; }
        public string CouponCode { get; set; }
        public string OrderStatus { get; set; }
    }
}
